app/models/version.rb

```diff
@@ -256,6 +262,22 @@ def authors_array
     self.authors.split(',').flatten
   end
 
+  def recalculate_sha256
+    key = "gems/#{full_name}.gem"
+    digest = Digest::SHA2.new
+    dir = Indexer.new.directory
+
+    dir.files.get(key) do |chunk|
+      digest << chunk
+    end
+
+    digest.base64digest
```

Comment:
Also, I kinda dislike this method has a different code-path that the one `pusher` runs... this generation should use the same code path, for the migration and for future generations!